modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java

```diff
@@ -335,6 +336,17 @@ else if (object instanceof Map) {
 			variables.putAll(
 				(Map<String, Object>)payloadJSONObject.get(
 					"extendedProperties"));
+
+			if (objectDefinitionName.equals("User")) {
+				if (variables.containsKey("firstName")) {
+					variables.put("givenName", variables.get("firstName"));
+				}
+
+				if (variables.containsKey("middleName")) {
+					variables.put(
+						"additionalName", variables.get("middleName"));
+				}
+			}
```

Comment:
@pedro-oliveira446 I'm not a big fan of adding something so specific for the User system object in a method that should be generic. Instead, I would create a default implemenation of `getVariables` in the `SystemObjectDefinitionMetadata` interface:

```
public default Map<String, Object> getVariables(JSONObject payloadJSONObject, ...) {
    ....
}
```
and I would change the implementation in the `UserSystemObjectDefinitionMetadata`:

```
@Override
public void getVariables(JSONObject payloadJSONObject, ...) {

	 Map<String, Object> variables = super.getVariables(payloadJSONObject, ...);
     
    // custom implementation
}
```
And here you can do:

```
SystemObjectDefinitionMetadata systemObjectDefinitionMetadata =
	_systemObjectDefinitionMetadataRegistry.
		getSystemObjectDefinitionMetadata(objectDefinition.getName());

Map<String, Object> variables = systemObjectDefinitionMetadata.getVariables(payloadJSONObject, ...);
```
@p-albuquerque Could you help Pedro with this issue? The solution doesn't have to be the one I suggested, but it would be good to keep in mind what I said at the beginning.

Reply:
Copy that